### ISSUE TITLE
Details on creating linear probe dataset

### ISSUE THREAD

**Issue by nguyentr17** [yuzhaouoe/SAE-based-representation-engineering#1]
Hi,

Thanks for the work and sharing the code! I'd like to ask how you created the conflict and non-conflict instances to train linear probes. When creating the train activation set, do you use exactly the same questions for all the instances and only change the evidence to either conflict with or align with the memorized knowledge? Or are the two conflict / no-conflict groups of instances created using different set of questions?

**Comment by yuzhaouoe** [comment]
> Hi,
> 
> Thanks for the work and sharing the code! I'd like to ask how you created the conflict and non-conflict instances to train linear probes. When creating the train activation set, do you use exactly the same questions for all the instances and only change the evidence to either conflict with or align with the memorized knowledge? Or are the two conflict / no-conflict groups of instances created using different set of questions?

Hi, thanks for your interest! The process of creating probing datasets is:

1. Select the questions the model can answer correctly without providing evidence (original answer's EM =1), which ensures the memorised knowledge aligns with the non-conflit context:  [code](https://github.com/yuzhaouoe/SAE-based-representation-engineering/blob/9008a0593e7715e45a597b121a1c7c195f3d8231/spare/kc_probing.py#L75)
2. Because there are duplicate questions in NQSwap (same question but different context), we avoid the same question appearing in the training and test sets: [code](https://github.com/yuzhaouoe/SAE-based-representation-engineering/blob/9008a0593e7715e45a597b121a1c7c195f3d8231/spare/kc_probing.py#L74)
3. We also avoid the same question appearing in conflict and non-conflict subsets of training set: [code](https://github.com/yuzhaouoe/SAE-based-representation-engineering/blob/9008a0593e7715e45a597b121a1c7c195f3d8231/spare/kc_probing.py#L86) and [code](https://github.com/yuzhaouoe/SAE-based-representation-engineering/blob/9008a0593e7715e45a597b121a1c7c195f3d8231/spare/kc_probing.py#L89)

**Comment by nguyentr17** [comment]
Thanks for your quick response! Why do we need to avoid the same question appearing in conflict and non-conflict subsets of training set? I thought we would want to construct the training set such that the only difference between conflict and no-conflict groups is the difference in in-context evidence so we would want to use the same set of questions for those 2 groups.

**Comment by yuzhaouoe** [comment]
> Thanks for your quick response! Why do we need to avoid the same question appearing in conflict and non-conflict subsets of training set? I thought we would want to construct the training set such that the only difference between conflict and no-conflict groups is the difference in in-context evidence so we would want to use the same set of questions for those 2 groups.

The target is not to classify a pair-wise conflict and non-conflict evidence for a question, and we want to know whether the model can classify arbitrary evidence for a question. Avoiding the same question appearing in conflict and non-conflict increases the difficulty and can result in a lower accuracy in the test set. It does not affect the conclusion if we do not avoid it, where the conflict is still more detectable at mid-layers.

**Comment by nguyentr17** [comment]
Thanks for the explanation! This has been very helpful.
